### ISSUE TITLE
Feature Request (Webtorrent): Provide option to "always" open torrent/magnet files with system default

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#6291]
## Description 

As it is now, users can disable Webtorrent in settings and after doing this, the browser will prompt the user after clicking/downloading on a magnet link or torrent file, asking whether or not you'd like to use [System default torrent client] to open the file.

![image](https://user-images.githubusercontent.com/3299098/66083314-dd897c80-e520-11e9-9617-0ddafd76a896.png)

However, with this method, users are prompted **every time** they attempt to download a torrent/magnet file. This can be cumbersome and there should be a way to tell the browser to always default to the users preferred torrent client.

**Consideration and Suggestion**
When adding this feature, there should also be a way to reverse the decision so that users can choose different clients if necessary. 

I would suggest adding a couple checkboxes to the UI for the simplest implementation (plz excuse the hasty shop job):
![ask](https://user-images.githubusercontent.com/3299098/66086651-89cf6100-e529-11e9-895f-d933d7b60ac8.png)

And in `Settings --> Extensions`:
![wt](https://user-images.githubusercontent.com/3299098/66086680-a5d30280-e529-11e9-9990-44c42e304ee8.png)

cc @yrliou @rebron

**Comment by bencefalo** [comment]
It would be great if when this is fixed its fixed for other apps and not just torrent... like zoom.us and webex. You are prompted every time to open zoom when browser from a zoom meeting

**Comment by Kartereo** [comment]
Yes please, we shouldn't have to click yes or no every single time we download a torrent.

**Comment by kartsims** [comment]
Weirdly, one of my Brave sessions (as in "signed-in as another user") provides this feature. When I click a torrent, it directly opens it with my default app. 

I don't know how to apply this option to the other session, but it _should_ be possible.

**Comment by Brave-Matt** [comment]
I have a user reporting the same situation, but its related to different external apps (in this users case, Roblox). I think it's worth extending this particular option to include any/all external apps.
https://www.reddit.com/r/brave_browser/comments/ecvgjb/this_makes_me_wanna_switch_back_to_chrome/

**Comment by bravesucks1** [comment]
DO NOT USE BRAVE BROWSER, if you use torrents. 

Utorrent and Brave browser do not and will not work together, and if you have them installed Brave browser will automatilly attach itself it Utorrent and then BLOCK IT. 

Save yourself the time and dont use "Brave Browser", promoted as a help its been nothing but crap.

**Comment by Brave-Matt** [comment]
Another +1 from a user opening MS Excel files:
https://www.reddit.com/r/brave_browser/comments/es8vkt/is_there_any_way_of_avoiding_this_prompt_every/

**Comment by Brave-Matt** [comment]
+1 https://github.com/brave/brave-browser/issues/6291

**Comment by Sntl09** [comment]
Yeah, sorry but this is where Brave lost me and i've uninstalled. It seemed like a great browser but the torrenting issue is a deal breaker for me. +1 to sort this

**Comment by eloquentvogon** [comment]
I use Zoom multiple times a day and I too need a "Don't ask again" option

**Comment by Brave-Matt** [comment]
Very similar +1:
https://community.brave.com/t/i-need-help-with-a-thing/107346

**Comment by Brave-Matt** [comment]
Another similar +1: https://www.reddit.com/r/brave_browser/comments/f3d78s/how_can_i_stop_brave_from_still_asking_if_it/

**Comment by Brave-Matt** [comment]
+1 from Community:
https://community.brave.com/t/remember-my-choice-when-websites-want-to-open-applications/114589

**Comment by michelle911** [comment]
+1, I too am forced to use another browser that allows this functionality, but like Brave better - please fix!

**Comment by pratyushtewari** [comment]
With more apps using the web url mechanism to open the desktop apps, e.g. Zoom, hangouts, download manager, email app, etc please consider "always allow" or "remember this app" feature.

**Comment by Brave-Matt** [comment]
+1 from Reddit:
https://www.reddit.com/r/brave_browser/comments/g09f6a/how_do_i_get_brave_to_stop_asking_for_permission/

**Comment by eljuno** [comment]
+1 from Community

https://community.brave.com/t/configure-brave-to-use-transmission-for-magnet-links/120235?u=eljuno

**Comment by benwoodward** [comment]
+1 

I have this issue with ["Add to Things 3"](https://chrome.google.com/webstore/detail/add-to-things-3/kakcfbhogeepoebcaidchmhphjeknpne?hl=en)

![image](https://user-images.githubusercontent.com/1472981/80148740-99b7ab00-8609-11ea-8b8c-eb0770bd71b7.png)

**Comment by Xan-Kun** [comment]
+1 No Skype for Business = No Brave
Sadly

**Comment by ramonsmits** [comment]
Zoom, Torrents, Git, etc. it all is *SO* annoying that I can't just select "Don't ask again for this domain".

**Comment by ramonsmits** [comment]
Also, can the title be adjusted? Its not only for Torrents but for all applications.

**Comment by ramonsmits** [comment]
Title suggestion

> Feature Request: Support `ExternalProtocolDialogShowAlwaysOpenCheckbox` similarly as Edge and Chrome

https://cloud.google.com/docs/chrome-enterprise/policies/?policy=ExternalProtocolDialogShowAlwaysOpenCheckbox

For now, I'm saying goodbye to Brave and hello Chrome until this policy is supported.

**Comment by bsclifton** [comment]
@ramonsmits the group policy should be supported 100% - you can try via registry. More info at https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy

We don't have proper group policy templates (ADM/ADMX) files available yet though

**Comment by ramonsmits** [comment]
@bsclifton Thanks for mentioning, searching via Google didn't contain these pages. THis is because they do not mention `ExternalProtocolDialogShowAlwaysOpenCheckbox `. Based on your comment the following registry file should allow to restore the checkbox again.

```reg
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Policies\BraveSoftware\Brave]
"ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:41000000
```

**Comment by markstuart** [comment]
So, you can get this to work on Linux right now... following the info from @bsclifton and @ramonsmits as well as this: https://www.chromium.org/administrators/linux-quick-start I got to this:
```bash
sudo mkdir -p /etc/chromium/policies/managed
sudo chmod -w /etc/chromium/policies/managed
sudo touch /etc/chromium/policies/managed/managed_policies.json
```
Edit the file using whatever you like (I used nano), and insert this json:
```json
{
  "ExternalProtocolDialogShowAlwaysOpenCheckbox": true
}
```
Go to brave://policy in the Brave address bar, and you should see this policy displaying.
Then if you click on a Zoom link or whatever, you should get the checkbox, and not get asked next time.

I guess Brave will possibly stop sharing the config directory with chromium at some point? This may be a short term fix.

**Comment by belminf** [comment]
Linux one-liner that worked for me:

```
sudo mkdir -p /etc/chromium/policies/managed && echo '{"ExternalProtocolDialogShowAlwaysOpenCheckbox": true}' | sudo tee /etc/chromium/policies/managed/protocol_open.json
```

No need to restart. Just needed to select the checkbox the first time opening a protocol. For e.g., for magnet links:

![image](https://user-images.githubusercontent.com/1307755/81697908-5ba4ee80-9433-11ea-87d0-13c177d3873d.png)

**Comment by dasforsyth** [comment]
> @bsclifton Thanks for mentioning, searching via Google didn't contain these pages. THis is because they do not mention `ExternalProtocolDialogShowAlwaysOpenCheckbox `. Based on your comment the following registry file should allow to restore the checkbox again.
> 
> ```reg
> Windows Registry Editor Version 5.00
> 
> [HKEY_CURRENT_USER\Software\Policies\BraveSoftware\Brave]
> "ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:41000000
> ```

I found the above didn't work for me, but the below did:

```Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
"ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
```

**Comment by kmf** [comment]
For MacOS
`defaults write com.brave.Browser ExternalProtocolDialogShowAlwaysOpenCheckbox -bool true`

**Comment by Xan-Kun** [comment]
@dasforsyth I can confirm ;-) Works for me too! *hooray*

**Comment by lilleyen** [comment]
> Linux one-liner that worked for me:
> 
> ```
> sudo mkdir -p /etc/chromium/policies/managed && echo '{"ExternalProtocolDialogShowAlwaysOpenCheckbox": true}' | sudo tee /etc/chromium/policies/managed/protocol_open.json
> ```
> 
> No need to restart. Just needed to select the checkbox the first time opening a protocol. For e.g., for magnet links:
> 
> ![image](https://user-images.githubusercontent.com/1307755/81697908-5ba4ee80-9433-11ea-87d0-13c177d3873d.png)

This worked perfectly for me. Linux Mint 19.3 Thank you, thank you!!

**Comment by Brave-Matt** [comment]
We have a Windows user here who seems to have a found a solution to the issue as well:
https://community.brave.com/t/unable-to-remove-confirmation-dialog-on-external-protocol-handler/120826

**Comment by fharper** [comment]
@kmf solution works for me on macOS, thanks. Note that you need to restart the browser.

**Comment by BryanChung** [comment]
As of now, I cannot find those Windows registry entries in my system. not sure why. I am on Brave V1.10.97 on Windows 10. Any idea why?

**Comment by bsclifton** [comment]
@BryanChung you'd have to create the registry structure for them to show up

As shared above, you can make a new file called `brave.reg` (or similar) and put the following in there:
```
[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
"ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
```

After saving that, you can double click the `.reg` file and it'll add the entry. Let us know if that solves the issue

**Comment by BryanChung** [comment]
> @BryanChung you'd have to create the registry structure for them to show up
> 
> As shared above, you can make a new file called `brave.reg` (or similar) and put the following in there:
> 
> ```
> [HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
> "ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
> ```
> 
> After saving that, you can double click the `.reg` file and it'll add the entry. Let us know if that solves the issue for you 😄

Thank you. I received this error:

![2020-06-30_114821](https://user-images.githubusercontent.com/67574157/86081328-af44c780-bac7-11ea-9cfc-7625fc144ee6.png)

![2020-06-30_114916](https://user-images.githubusercontent.com/67574157/86081351-bd92e380-bac7-11ea-8df0-61c4a1f637af.png)

Any idea why?

**Comment by apennismightier** [comment]
> > @BryanChung you'd have to create the registry structure for them to show up
> > As shared above, you can make a new file called `brave.reg` (or similar) and put the following in there:
> > ```
> > [HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
> > "ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
> > ```
> > 
> > 
> > After saving that, you can double click the `.reg` file and it'll add the entry. Let us know if that solves the issue for you smile
> 
> Oh! I needed to create one rather than finding it. Haha! Ok thanks!

That directory doesn't exist in either the HKEY_LOCAL_MACHINE or the HKEY_CURRENT_USER for me. I'm on Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)

**Comment by BryanChung** [comment]
> > > @BryanChung you'd have to create the registry structure for them to show up
> > > As shared above, you can make a new file called `brave.reg` (or similar) and put the following in there:
> > > ```
> > > [HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
> > > "ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
> > > ```
> > > 
> > > 
> > > After saving that, you can double click the `.reg` file and it'll add the entry. Let us know if that solves the issue for you smile
> > 
> > 
> > Oh! I needed to create one rather than finding it. Haha! Ok thanks!
> > @BryanChung @bsclifton
> 
> That directory doesn't exist in either the HKEY_LOCAL_MACHINE or the HKEY_CURRENT_USER for me. I'm on:
> 
> Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)
> Windows 8.1

Yeah we need to create the directory. :) You can use mine.

[brave.zip](https://github.com/brave/brave-browser/files/4894728/brave.zip)

Don't blame me if it breaks. :P

**Comment by NRN1337** [comment]
> > > > @BryanChung you'd have to create the registry structure for them to show up
> > > > As shared above, you can make a new file called `brave.reg` (or similar) and put the following in there:
> > > > ```
> > > > [HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave]
> > > > "ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:1
> > > > ```
> > > > 
> > > > 
> > > > After saving that, you can double click the `.reg` file and it'll add the entry. Let us know if that solves the issue for you smile
> > > 
> > > 
> > > Oh! I needed to create one rather than finding it. Haha! Ok thanks!
> > > @BryanChung @bsclifton
> > 
> > 
> > That directory doesn't exist in either the HKEY_LOCAL_MACHINE or the HKEY_CURRENT_USER for me. I'm on:
> > Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)
> > Windows 8.1
> 
> Yeah we need to create the directory. :) You can use mine.
> 
> [brave.zip](https://github.com/brave/brave-browser/files/4894728/brave.zip)
> 
> Don't blame me if it breaks. :P

Doesn't work for me.

**Comment by Xan-Kun** [comment]
> That directory doesn't exist in either the HKEY_LOCAL_MACHINE or the HKEY_CURRENT_USER for me. I'm on:
> 
> Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)
> Windows 8.1

Yeah, that's why it would have been so important for everybody to post their exact OS :-/ The paths are different, obviously.

Thanks for being a good example! :-)

**Comment by Bargsteen** [comment]
The current solution posted here only works on a per-site basis. It would be great if you could remove the prompt for a specific app for all sites. 
Context: I am using Org Capture in Emacs to save bookmarks to a file on disk. This is a reasonably common use case for Emacs users :)

**Comment by aspiers** [comment]
@Bargsteen I have posted a solution for exactly that [here](https://superuser.com/a/1588146/104997) :-)

**Comment by 4evermaat** [comment]
I don't understand why this is an "issue" that hasn't been resolved with brave 

![Snag_240d27b2](https://user-images.githubusercontent.com/830306/100496484-f4f4e180-3186-11eb-87d8-c0b1d64f2bab.png)


A lot of brave versions have come out since this was first reported.   Where is the checkbox that can store the user's preference?   [I prefer to save the `.torrent` file]

**Comment by LanceHaverkamp** [comment]
This also affects mailto, which is use by **_far_** more people than use torrents.

**Comment by SpyderZ** [comment]
I can see this for the first time used for a 3rd party extension, but after that, it should never happen again.

**Comment by jtoy** [comment]
+1

**Comment by cprance** [comment]
+1

**Comment by alik604** [comment]
I finally forced myself to use Brave full time, and now im waiting for firefox to use groups... (or I just use a ungoogled-chromium)

This is a horrifying. I insist on using my qBittorrent which is linked to my VPN interface/adaptor. 

`Settings --> Extensions` does not have webtorrent

**Comment by alik604** [comment]
id much rather use my qBittorrent which is linked to my VPN interface/adaptor. This is a liability, maybe even _malware_ as it carries a risk of getting me a fine if I act carelessly.

**`Settings --> Extensions` does not have webtorrent, its been moved to `Settings` - just use the search function. I was able to auto use my default client, qBittorrent.** problem solved for me - on windows 10

**Comment by aspiers** [comment]
[**@Xan-Kun**](https://github.com/Xan-Kun) commented on [July 1, 2022 3:03 PM](https://github.com/brave/brave-browser/issues/6291#issuecomment-1172383619 "2022-07-01T14:03:13Z - Replied by Github Reply Comments"):
> Good bye Brave. This is as long as I was willing to wait.

What are you waiting for exactly?  There are workarounds posted in this issue.

**Comment by SpyderZ** [comment]
> [**@Xan-Kun**](https://github.com/Xan-Kun) commented on [July 1, 2022 3:03 PM](https://github.com/brave/brave-browser/issues/6291#issuecomment-1172383619):
> 
> > Good bye Brave. This is as long as I was willing to wait.
> 
> What are you waiting for exactly? There are workarounds posted in this issue.

Certain extensions constantly request permission to open a 3rd party app for each domain. It's totally annoying and needs to be adjusted to allow global permissions for all domains after permission is granted the first time.
![Screen Shot 2022-07-02 at 6 05 19 PM](https://user-images.githubusercontent.com/1954540/177018514-e70996a3-7832-42c4-8142-472bb9290eef.png)

**Comment by aspiers** [comment]
The solution for that [above](https://github.com/brave/brave-browser/issues/6291#issuecomment-718764271) has worked great for me for several months on Brave / Chrome / Chromium.

**Comment by kerim** [comment]
I'm having this problem as well (and don't really understand the solution linked above). Why can't I just permit opening an app across ALL websites once and never have to think about it again?

**Comment by benesjan** [comment]
Why has this not been solved yet without having to hack around? Who wants to use shitty web apps when you have fast native ones which are actually integrated with the OS?

**Comment by kj4ezj** [comment]
+1

I have a unique use case for which I have yet to find a workaround.

## Problem
I have to use Zoom video conferencing on my personal computer (Linux Mint) for work, school, and sometimes even family. Zoom has shown the world that it is not safe to run their native client on your computer<sup>[[1](https://www.tomsguide.com/news/zoom-ftc-settlement), [2](https://www.tomsguide.com/news/zoom-security-privacy-woes), [3](https://arstechnica.com/information-technology/2022/08/zoom-patches-critical-vulnerability-again-after-prior-fix-was-bypassed/), [4](https://www.zdnet.com/article/zoom-defends-use-of-local-web-server-on-macs-after-security-report/)]</sup> because of negligence or malice, so I use it exclusively in a web browser. Video conferencing tends to "just work" in Chromium-family browsers, so I use Brave for Zoom.

When you open a meeting link, it tries to invoke their native app whether you have it or not.

![2022-10-11 18-21-51 - Brave Zoom prompt with missing link](https://user-images.githubusercontent.com/34947245/195214192-52ba35bf-441d-46eb-b45c-651c642f3d2c.png)

You may notice the `Join from Your Browser` link is missing. This doesn't _always_ happen, but it usually does. So, once I click `Cancel` on the `Open xdg-open` prompt from Brave, I have to click `Launch Meeting`, and deal with the prompt **again**!

![2022-10-11 18-22-32 - Brave Zoom second prompt](https://user-images.githubusercontent.com/34947245/195214599-ac6ddd7a-ef4e-470f-9233-35e52b411f1d.png)

Once Zoom has prompted my browser twice, it will always show the `Join from Your Browser` option.

As upset as your community members are for having to deal with this prompt _once_ multiple times every single day, imagine how excited I am to have to deal with it _TWICE_ multiple times every singe day. :)

## ~~Solution~~ What I Tried
I spent a few hours on this, read a bunch, and tried a number of things. Ultimately I found [this Stack Overflow answer](https://stackoverflow.com/a/64538468), [this article](https://best-mac-tips.com/2022/04/23/disable-open-external-app-prompt-in-chrome-and-brave-browser/) for macOS that gave me the protocol URI Zoom uses, and [this Stack Exchange answer](https://superuser.com/a/1588146) @aspiers shared earlier in this thread which pointed me in the right direction to figure out where these policies live on Linux these days, because they have moved around.

The closest I got to a working solution was this.
```bash
sudo mkdir -p /etc/brave/policies/managed
jq -n '.URLBlocklist=["zoommtg:*","zoommtg://*"]' | sudo tee /etc/brave/policies/managed/blocklist.json
```
I also threw in the policy to force the "Always allow" checkbox to appear, even though Zoom already has it, for good measure.
```bash
sudo mkdir -p /etc/brave/policies/managed
jq -n '.ExternalProtocolDialogShowAlwaysOpenCheckbox=true' | sudo tee /etc/brave/policies/managed/protocol_open.json
```
These policies were picked up by Brave!

![2022-10-11 18-23-32 - Brave Zoom policies](https://user-images.githubusercontent.com/34947245/195216779-a15904c9-7c4a-4e8b-86ef-52915e256232.png)

Unfortunately, instead of blocking the Zoom website from invoking the `xdg-open` dialogue box for `zoommtg://*` URIs and letting me use the browser, it prevented the page from loading entirely.

![2022-10-11 18-23-57 - Brave Zoom blocked](https://user-images.githubusercontent.com/34947245/195216971-0a6f65cf-2ef4-4bce-99d5-22c3f4cc0c0c.png)

I guess the prompts are technically gone, but this did not solve my problem so I reverted the blocklist.

## System
System
```
Kernel: 5.4.0-126-generic x86_64 bits: 64 compiler: gcc v: 9.4.0 
Desktop: Cinnamon 5.2.7 wm: muffin dm: LightDM Distro: Linux Mint 20.3 Una 
base: Ubuntu 20.04 focal
```
Brave
```
Version 1.44.108 Chromium 106.0.5249.103 (Official Build) (64-bit)
```

## 

Please add some mechanism to block specific sites from invoking this pop-up, or automatically deny it. If you read the original issue, OP asked for two things (admittedly with an "or") and there is a workaround for some use-cases to accomplish one of them, but a lot of people asking for the other with no workarounds I could find. I don't care if it is a checkbox, button, policy, flag, or a seance I conduct by putting candles and salt in a pattern around my computer under a full moon...but we need something.

I am not able to PR a fix myself without a ton of hand-holding, but I can help test solutions and open to other ideas from the community. Thank you for your time and product, I'll be a Brave user five years this December!

**Comment by asheroto** [comment]
Possible duplicate of:

#16927
#26058

**Comment by wiesl** [comment]
Doesn't work for me with Brave Version 1.68.137 Chromium: 127.0.6533.100 (Official Build) (64-bit) on Windows 10.

I'm using the following extension with assigned hot key: Share link via email
https://chromewebstore.google.com/detail/share-link-via-email/ejdbkikfbnnhmachnnomjfgjbgkcnjkb

But I'm still getting the following popup (even with registry entries below and checking the always allow option):
chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb wants to open this application.
[ ] Always allow chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb to open links of this type in the associated app

Any ideas?

Thnx.

```
REGEDIT4

[HKEY_CURRENT_USER\Software\Policies\BraveSoftware\Brave]
"ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:00000001

[HKEY_CURRENT_USER\SOFTWARE\BraveSoftware\Brave-Browser]
"ExternalProtocolDialogShowAlwaysOpenCheckbox"=dword:00000001
```

**Comment by asheroto** [comment]
`ejdbkikfbnnhmachnnomjfgjbgkcnjkb ` looks to be [Share link via email](https://chromewebstore.google.com/detail/share-link-via-email/ejdbkikfbnnhmachnnomjfgjbgkcnjkb), right? If you disable that extension do you still have the same issue?

Just trying to help isolate the problem. 😊

**Comment by wiesl** [comment]
Yes, as I wrote this is Share link via email.

So I configured the extension with an assigned hotkey (in my case Ctrl-Y).

Testcase:
1.) Press Ctrl-Y on a webpage (should just open the default E-Mail program with the link in an email)
2.) Getting the following popup
chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb wants to open this application.
[ ] Always allow chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb to open links of this type in the associated app
Buttons: Button 1: Open Mozilla Thunderbird, Portable Edition, Button 2: Cancel
3.) I press Open Mozilla Thunderbird, Portable Edition and the E-Mail will be opened.
(Checking "Always allow chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb to open links of this type in the associated app" doesn't have any effect) 

Step 2 and 3 are the problem, shouldn't appear.

Should be easy to be reproduceable.

Doesn't appear in any other context. So disabling will not have any effect as I can't test it then 😊

**Comment by asheroto** [comment]
@wiesl sorry, my fault, I misread your post.

I am also experiencing a similar issue. The original post was made almost 5 years ago. 😊 I [posted about this](https://github.com/brave/brave-browser/issues/16927#issuecomment-1662734907) a year ago as well.

The developers must be busy with other things, I guess.

---

The OP mentioned `qBittorrent` but that is when the `magnet:` link is used. You can fix this by creating a registry key in the `URLAllowList`

Brave reads Chrome registry keys as well, so I tried this:

```reg
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Policies\Google\Chrome\URLAllowList]
"1"="mailto://*"
"2"="tel://*"
"3"="magnet://*"
```

And reloaded Brave with `brave://restart` then `tel:` hyperlinks launch the application directly and well as `magnet:` hyperlinks, but `mailto:` hyperlinks continue to show the popup.

It seems that the issue seems to stem from the browser not properly recognizing the `protocol_handler` -> `allowed_origin_protocol_pairs` key in the `Preferences` file. Although selecting the "Always allow" checkbox does add the entry to this key, the browser is not actually processing it correctly. I'm not sure though, just a strong guess.

I've also tried [this fix](https://stackoverflow.com/a/63557253/11404332) but it did not work.

**Comment by TheOverpassArsonist** [comment]
It's not the part that's getting the most attention here, but I'd also specifically want to mention the "Ask which app brave should use for torrents" option would also fix some really annoying problems. For some reason stremio got set as my default torrent client or something and now no matter what I try brave keeps trying to open magnet links in stremio as opposed to an actual torrent client and, since magnet links aren't actual files, I can't find anyway to change this association on an OS level. (Linux if relevant)

**Comment by DDQW** [comment]
brave stopped opening magnet links for me again (it was opening them with qbittorrent just fine for a while) i don't think it was a good idea to scropecreep with webtorrent and basically breaking it for everything else

**Comment by Stebalien** [comment]
You can work around this by adding the following top-level key/value to the `Preferences` object when brave isn't running.

```json
  "protocol_handler": {
    "policy": {
      "auto_launch_protocols_from_origins": [
        {
          "allowed_origins": [
            "ALLOWED_ORIGIN"
          ],
          "protocol": "ALLOWED_PROTOCOL"
        }
      ]
    }
  }
```

Where `ALLOWED_ORIGIN` is the origin that's allowed to launch the protocol handler of `ALLOWED_PROTOCOL`.

See https://github.com/brave/chromium/blob/4e22a93b12f76cc59ee084f7f6887afc232e2afd/chrome/common/pref_names.h#L2100-L2106.

**Comment by wiesl** [comment]
> You can work around this by adding the following top-level key/value to the `Preferences` object when brave isn't running.
> 
>   "protocol_handler": {
>     "policy": {
>       "auto_launch_protocols_from_origins": [
>         {
>           "allowed_origins": [
>             "ALLOWED_ORIGIN"
>           ],
>           "protocol": "ALLOWED_PROTOCOL"
>         }
>       ]
>     }
>   }
> Where `ALLOWED_ORIGIN` is the origin that's allowed to launch the protocol handler of `ALLOWED_PROTOCOL`.
> 
> See https://github.com/brave/chromium/blob/4e22a93b12f76cc59ee084f7f6887afc232e2afd/chrome/common/pref_names.h#L2100-L2106.

@Stebalien

Thnx.
Any actual sample values for ALLOWED_ORIGIN or ALLOWED_PROTOCOL?

e.g. for the following extension:
Share link via email
https://chromewebstore.google.com/detail/share-link-via-email/ejdbkikfbnnhmachnnomjfgjbgkcnjkb

**Comment by Stebalien** [comment]
Assuming the protocol is `mailto:`:

```
  "protocol_handler": {
    "policy": {
      "auto_launch_protocols_from_origins": [
        {
          "allowed_origins": [
            "chrome-extension://ejdbkikfbnnhmachnnomjfgjbgkcnjkb"
          ],
          "protocol": "mailto"
        }
      ]
    }
  }
```

**Comment by wiesl** [comment]
@Stebalien 

Thnx.

Tried it, but it didn't work.

For reference: Preferences files is under Windows OS at: "C:\Users\username\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default\Preferences" 

Can you maybe try it, too?
Any further ideas?

**Comment by J-Wick4** [comment]
Attached is what I used on MacOS to resolve this problem. Cheers.

[How to Stop the Damn Brave Browser Local Application Pop-ups.pdf](https://github.com/user-attachments/files/21264035/How.to.Stop.the.Damn.Brave.Browser.Local.Application.Pop-ups.pdf)

**Comment by Stebalien** [comment]
@wiesl that's the right file.

1. Make sure to edit it when brave _isn't_ running (even in the background).
2. Make sure to add the key/value to the top-level of the object.

Other than that, maybe there's something special about "mailto" or the extension in question? My issue was with a different extension/protocol.

**Comment by wiesl** [comment]
> [@wiesl](https://github.com/wiesl) that's the right file.
> 
> 1. Make sure to edit it when brave _isn't_ running (even in the background).
> 2. Make sure to add the key/value to the top-level of the object.
> 
> Other than that, maybe there's something special about "mailto" or the extension in question? My issue was with a different extension/protocol.

@Stebalien 

Yes, closed Brave (killed all processes) and added it to the top-level.

Can you reproduce it?

Thnx.
